File: 2025/go/benchmark1/performance.go
package main

import (
	"fmt"
	"math"
	"time"
)

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	sqrt := int(math.Sqrt(float64(n)))
	for i := 2; i <= sqrt; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func countPrimes(limit int) int {
	count := 0
	for i := 2; i <= limit; i++ {
		if isPrime(i) {
			count++
		}
	}
	return count
}

func main() {
	n := 1000000 // Define a large number
	start := time.Now()
	result := countPrimes(n)
	elapsed := time.Since(start)
	fmt.Printf("Number of primes: %d\n", result)
	fmt.Printf("Execution time (Go): %.4f seconds\n", elapsed.Seconds())
}

File: 2025/go/benchmark1/performance.rs
use std::time::Instant;

fn is_prime(n: u32) -> bool {
    if n < 2 {
        return false;
    }
    let sqrt_n = (n as f64).sqrt() as u32;
    for i in 2..=sqrt_n {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn count_primes(limit: u32) -> u32 {
    let mut count = 0;
    for n in 2..=limit {
        if is_prime(n) {
            count += 1;
        }
    }
    count
}

fn main() {
    let n: u32 = 1_000_000; // Define a large number
    let start = Instant::now();
    let result = count_primes(n);
    let elapsed = start.elapsed();
    println!("Number of primes: {}", result);
    println!("Execution time (Rust): {:.4} seconds", elapsed.as_secs_f64());
}
